vllm/model_executor/models/utils.py

```diff
@@ -231,6 +231,88 @@ def load_weights(
         return autoloaded_weights
 
 
+class OffloadedTensor(torch.Tensor):
+
+    @staticmethod
+    def __new__(cls, elem, *, requires_grad=None):
+        # the wrapped tensor will have the same
+        # metadata as the original tensor
+        if elem.device != torch.device("cpu"):
+            # do not hold a strong reference to the tensor
+            elem = weak_ref_tensor(elem)
+        if requires_grad is None:
+            return super().__new__(cls, elem)
+        else:
+            return cls._make_subclass(cls, elem, requires_grad)
```

Comment:
I think if your tensor subclass internally holds another tensor (`elem` here), you probably want to user the "wrapper" subclass API. Example [here](https://github.com/pytorch/pytorch/blob/main/torch/_functorch/aot_autograd.py)
```
out_tensor = torch.Tensor._make_wrapper_subclass(cls, shape, **kwargs)
out_tensor.elem = weak_ref_tensor(elem)
```

Side note: I would probably call that constructor unconditionally, any reason you aren't doing it when `requires_grad is None`?

Reply:
I copy the code from https://github.com/albanD/subclass_zoo/blob/ec47458346c2a1cfcd5e676926a4bbc6709ff62e/base_tensor.py#L11